Tests/SwiftParserTest/DirectiveTests.swift

```diff
@@ -184,4 +184,29 @@ final class DirectiveTests: XCTestCase {
     )
   }
 
+  func testFollowedByDeclarations() {
+    assertParse(
+      """
+      struct Foo {
+        #if false
+        var x: Int
+      #endif1️⃣; var x = 1
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          message: "extra tokens following conditional compilation directive",
+          fixIts: ["insert newline"]
+        )
+      ],
+      fixedSource: """
+      struct Foo {
+        #if false
+        var x: Int
+      #endif
+      ; var x = 1
+      }
+      """
```

Comment:
Not sure if we should add this Fix-It.
The current parser doesn't do it. It only add an error

Reply:
I don’t think we should add the Fix-It because I don’t expect anyone to write a statement after `#endif` and expect it to work. The more realistic error case I would imagine is just some bogus left-over code like `#endif s` and inserting a newline isn’t the fix there. I’m not even sure if we need a custom diagnostic for it.